compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -589,7 +597,16 @@ impl<'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
         span: Span,
         res: Option<Res>,
         base_error: &BaseError,
-    ) -> (bool, Vec<ImportSuggestion>) {
+    ) -> (bool, FxHashSet<String>, Vec<ImportSuggestion>) {
+        let span = match following_seg {
+            Some(_) if path[0].ident.span.eq_ctxt(path[path.len() - 1].ident.span) => {
```

Comment:
Is the branch on `following_seq` necessary? Presumably the path-derived span would be the same as the input `span` but is that actually the case?

Reply:
[Macros](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=25367c71f98c1a1c4e6628d82696305a).

Reply:
It's always macros